src/parser/Lexer.cpp

```diff
@@ -2108,34 +2219,41 @@ static ALWAYS_INLINE KeywordKind isKeyword(const StringBufferAccessData& data)
 ALWAYS_INLINE void Scanner::scanIdentifier(Scanner::ScannerResult* token, char16_t ch0)
 {
     ASSERT(token != nullptr);
-    Token type;
+    Token type = Token::IdentifierToken;
     const size_t start = this->index;
 
     // Backslash (U+005C) starts an escaped character.
     ScanIDResult id = UNLIKELY(ch0 == 0x5C) ? this->getComplexIdentifier() : this->getIdentifier();
+    const auto& data = std::get<0>(id);
     const size_t end = this->index;
 
     // There is no keyword or literal with only one character.
     // Thus, it must be an identifier.
-    KeywordKind keywordKind;
-    const auto& data = std::get<0>(id);
-    if (data.length == 1) {
-        type = Token::IdentifierToken;
-    } else if ((keywordKind = isKeyword(data))) {
-        token->setKeywordResult(this->lineNumber, this->lineStart, start, this->index, keywordKind);
-        return;
-    } else if (data.length == 4) {
-        if (data.equalsSameLength("null")) {
+    if (data.length > 1) {
+        KeywordKind keywordKind = getKeyword(data);
+
+        token->secondaryKeywordKind = keywordKind;
+
+        switch (keywordKind) {
+        case NotKeyword:
+            break;
+        case NullKeyword:
             type = Token::NullLiteralToken;
-        } else if (data.equalsSameLength("true")) {
-            type = Token::BooleanLiteralToken;
-        } else {
-            type = Token::IdentifierToken;
+            break;
+        case TrueKeyword:
+        case FalseKeyword:
+            type = BooleanLiteralToken;
+            break;
+        default:
+            if (keywordKind >= StrictModeReservedWord) {
+                break;
+            }
+            FALLTHROUGH;
+        case YieldKeyword:
```

Comment:
BTW why did you locate the `default` case in the middle of switch statement?
Just for readability?

Reply:
Becasue `YieldKeyword` is > StrictModeReservedWord. If I understand the standard correctly, only implements, interface, package, private, protected, and public are reserved in strict, yield and let are not. Probably there is some implementation detail why Escargot handles them as keywords. Maybe reorganizning the enum could help to change this.

Reply:
AKAIK the location of `default` does not matter.
In this case, switch would go into the default case at the last.
So, What about locating this default in the end as usual?